bound/bound.go

```diff
@@ -36,3 +35,22 @@ func Intersection(bounds ...Bound) Bound {
 
 	return ret
 }
+
+// Union returns a contiguous union of the input bounds if possible.
+// Otherwise a NaN Bound is returned.
+func Union(bounds ...Bound) Bound {
+	if len(bounds) == 0 {
+		return Bound{Min: math.NaN(), Max: math.NaN()}
+	}
+
+	ret := Bound{Min: bounds[0].Min, Max: bounds[0].Max}
+	for _, b := range bounds[1:] {
+		if ret.Max >= b.Min && b.Max >= ret.Min {
```

Comment:
If you invert this test, then the return takes the execution out of the function and we don't need an else block. It will also be clearer.
```
if b.Max < ret.Min || ret.Max < b.Min {
	return Bound{Min: math.NaN(), Max: math.NaN()}
}
ret.Min = math.Min(ret.Min, b.Min)
ret.Max = math.Max(ret.Max, b.Max)
```